website/members/models/member.py

```diff
@@ -117,6 +117,17 @@ def latest_membership(self):
             return None
         return self.membership_set.latest("since")
 
+    @property
+    def oldest_membership_of_current_type(self):
```

Comment:
I'd rather see this as part of the membership so that you can get that first connected membership of the same type from any membership, not just the latest one.

Reply:
I would say it's alike ```current_membership```, ```latest_membership```, ```earliest_membership```, which are all in Member

Reply:
It really is not, the filter is on the membership that you already have, not based on the user like those are. In all those cases you do not have a membership yet. You want to know 'the latest membership of the user' or the 'current membership of the user'.

Here you want "the first membership of the current membership's type"